octoprint_TimeToFilament/__init__.py

```diff
@@ -66,6 +79,33 @@ def on_startup(self, host, port):
   @octoprint.plugin.BlueprintPlugin.route("/get_settings_defaults", methods=["GET"])
   def get_settings_defaults_as_string(self):
     return json.dumps(self.get_settings_defaults())
+    
+  def is_blueprint_csrf_protected(self):
+    return True
+    
+  @octoprint.plugin.BlueprintPlugin.route("/", methods=["GET"])
+  def get_cache(self):
+    if self._printer.get_state_id() not in (["PRINTING", "PAUSED"]):
+      return flask.jsonify(None)
+    addData = self.additional_state_data(False)
+    dispLines = self._settings.get(["displayLines"])
+    for idx, line in enumerate(dispLines):
+      dispLines[idx] = dict_merge(line, addData[line["regex"]])
+      if "timeLeft" in dispLines[idx]:
+        dispLines[idx]["timeLeft"] = max(self._printer.get_current_data()["progress"]["printTimeLeft"]-dispLines[idx]["timeLeft"],0)
+        import datetime
+        dispLines[idx]["timeLeftFormatted"] = str(datetime.timedelta(seconds=dispLines[idx]["timeLeft"])).split('.', 2)[0]
```

Comment:
Above I use the "formatDuration" function.  Is that javascript, though, or python?  Maybe there is a python version, too?  Would that look nicer?